### PR TITLE
Add support for watchOS to the podspecs

### PR DESCRIPTION
**Pull request by daltonclaybrook**
grpc-swift is able to support watchOS 6+ with NIO Transport Services and Network.framework, but the podspecs do not declare such support, so it cannot be used in a watchOS app through CocoaPods. This PR adds the necessary support to each of the three podspecs.

### CODE REVIEW

**Review by glbrntt — CHANGES_REQUESTED** [review]
Thanks for the PR @daltonclaybrook -- the podspecs are generated by `/scripts/build_podspecs.py`, would you mind updating the script instead?

Also, I'm curious, have you actually tried gRPC on watchOS? It's something I've been meaning to do but haven't yet found the time for.

**Review by vinnyt — APPROVED** [review]

**Comment by daltonclaybrook** [comment]
@glbrntt I'd be happy to do that.
>have you actually tried gRPC on watchOS?

Yes! It works really well and I've been super pleased with it. [This was the crucial step for me to get it working](https://github.com/grpc/grpc-swift/blob/main/docs/apple-platforms.md). Though, it seems I may have one final hurdle getting grpc-swift to rebuild from Bitcode on watchOS. I've created #999 to track this issue. If you have any thoughts or suggestions on this, I would be immensely grateful.

**Comment by daltonclaybrook** [comment]
@glbrntt good for another look

**Review by glbrntt — APPROVED** [review]
Thanks @daltonclaybrook!

I just filed PRs for the various NIO components to make sure they also have watchOS deployment targets: 

- https://github.com/apple/swift-nio/pull/1679
- https://github.com/apple/swift-nio-http2/pull/251
- https://github.com/apple/swift-nio-ssl/pull/252 
- https://github.com/apple/swift-nio-transport-services/pull/107

It looks like SwiftLog and SwiftProtobuf are already good to go. I'll wait for the above to be merged before merging this PR.

**Comment by glbrntt** [comment]
> > have you actually tried gRPC on watchOS?
> 
> Yes! It works really well and I've been super pleased with it.

That's great to hear!


> [This was the crucial step for me to get it working](https://github.com/grpc/grpc-swift/blob/main/docs/apple-platforms.md). Though, it seems I may have one final hurdle getting grpc-swift to rebuild from Bitcode on watchOS. I've created #999 to track this issue. If you have any thoughts or suggestions on this, I would be immensely grateful.

I'm afraid not but it seems like you and @Lukasa have made progress on this.

**Pull request merged**
